framework/auth/core.py

```diff
@@ -1228,6 +1227,11 @@ def merge_user(self, user):
             user_settings.merge(addon)
             user_settings.save()
 
+        # Disconnect signal to prevent emails being sent about being a new contributor when merging users
+        # be sure to reconnect it at the end of this code block.
+        from website.project.signals import contributor_added
+        from website.project.views.contributor import notify_added_contributor
```

Comment:
I don't think there will be a circulated import problem here. Better try to avoid import like this as much as you can.
Try this
from website import project at the top
and contributor_added will be project.singals.contributor_added
and notify_added_contributor will be project.views.contributor.notify_Added_contributor